include/frg/logging.hpp

```diff
@@ -135,6 +135,42 @@ constexpr auto output_to(Container &cont) {
 	return container_logger{cont};
 }
 
+#ifdef FRIGG_HAS_IOSTREAM
+	template<typename Out>
+	struct ostream_out {
+		Out &output;
+
+		ostream_out(Out &out) : output(out) { }
+
+		void append(const char *str) {
+			output << str;
+		}
+
+		void append(char c) {
+			output << c;
+		}
+
+		auto &operator<<(auto &&t) {
+			format(std::forward<decltype(t)>(t), *this);
+			return *this;
+		}
+
+		auto &operator<<(endlog_t) {
+			output << std::endl;
+			return *this;
+		}
+
+		auto &operator<<(Out &(*os)(Out &)) {
```

Comment:
hmm, should iomanips be supported?  they wouldn't be able to have correct semantics on frigg-formatted in/outs since frigg dumbs down input data to characters or character arrays